exporter/clients.go

```diff
@@ -16,7 +16,7 @@ import (
 	id=11 addr=127.0.0.1:63508 fd=8 name= age=6321 idle=6320 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=setex
 	id=14 addr=127.0.0.1:64958 fd=9 name= age=5 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=26 qbuf-free=32742 obl=0 oll=0 omem=0 events=r cmd=client
 */
-func parseClientListString(clientInfo string) ([]string, bool) {
+func parseClientListString(clientInfo string) (map[string]string, bool) {
```

Comment:
I like that you're making this a bit more explicit and not use the array any longer but while you're changing this, maybe turn this into a `ClientInfo` struct.

Reply:
Also, I don't see where you're extracting `"user"` - I think that's missing.
White you're at it, want to add `resp` as well?

Reply:
* There is now a ClientInfo struct.

* "user" field is extracted as the rest of the fields during the `range strings.Split(clientInfo, " ")`

The clientinfo string is like `id=4 addr=127.0.0.1:50000 laddr=127.0.0.1:6379 fd=8 name= age=5 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=26 qbuf-free=40928 argv-mem=10 obl=0 oll=0 omem=0 tot-mem=61466 events=r cmd=client user=default redir=-1` from a redis>6 server

* Sorry, but I do not understand the _resp_ part

Please let me know anything, thanks for the feedback !

Reply:
Oh, the [documentation here says](https://redis.io/commands/client-list/) `resp: client RESP protocol version. Added in Redis 7.0`